app/org/sagebionetworks/bridge/services/AuthenticationService.java

```diff
@@ -238,6 +254,40 @@ public void resetPassword(PasswordReset passwordReset) throws BridgeServiceExcep
         accountDao.resetPassword(passwordReset);
     }
     
+    private void repairConsents(Study study, Account account, UserSession session, ClientInfo clientInfo){
+        boolean repaired = false;
+        
+        Map<SubpopulationGuid,ConsentStatus> statuses = session.getUser().getConsentStatuses();
+        for (Map.Entry<SubpopulationGuid,ConsentStatus> entry : statuses.entrySet()) {
+            ConsentSignature activeSignature = account.getActiveConsentSignature(entry.getKey());
+            ConsentStatus status = entry.getValue();
+
+            if (activeSignature != null && !status.isConsented()) {
+                repairConsent(session, entry.getKey(), activeSignature);
+                repaired = true;
+            }
+        }
+        
+        // These are incorrect since they are based on looking up DDB records, so re-create them.
+        if (repaired) {
+            updateConsentStatuses(study, session.getUser(), clientInfo);
+        }
+    }
+    
+    /**
+     * If the signature exists but consentStatus says the user is not consented, then the record is missing in
+     * DDB. We need to create it.
+     */
+    private void repairConsent(UserSession session, SubpopulationGuid subpopGuid, ConsentSignature activeSignature) {
+        logger.error("Signature found without a matching user consent record. Adding consent for " + session.getUser().getId());
```

Comment:
I'd like to know if this happens. I don't think we need to do anything about it, we'd just do what we're automating here. But I would like to look and see if users are still being created in this state, or if this is only a historical problem. That is, some users have had a DynamoUserConsent2 record, and some have not.